source/extensions/tracers/opentelemetry/samplers/dynatrace/sampler_config_provider.cc

```diff
@@ -47,8 +50,9 @@ SamplerConfigProviderImpl::SamplerConfigProviderImpl(
     } else {
       Http::RequestMessagePtr message = Http::Utility::prepareHeaders(http_uri_);
       message->headers().setReferenceMethod(Http::Headers::get().MethodValues.Get);
-      message->headers().setReference(Http::CustomHeaders::get().Authorization,
-                                      authorization_header_value_);
+      for (const auto& header_pair : parsed_headers_to_add_) {
+        message->headers().setReference(header_pair.first, header_pair.second);
+      }
```

Comment:
LGTM overall except this one. Basically, we think the token or passwd should be kept in secret. And if an raw string is used then we should mark it as `sensitive` and the `config_dump` will hide it when dump all the configurations.

So, if you use the `request_headers_to_add`, the token may be exposed in the dumped configuration file.

I you think it's OK, then I am also OK to this design.

Reply:
Thank's for the hint. I have discussed this with @joaopgrassi. We are aware of the problem but we would like to keep it as is for now. We would have to add the `[(udpa.annotations.sensitive) = true]` annotation to the `http_service`field. This would make it very hard to debug a wrong config.

Reply:
My suggestion  is to keep the `token` in another independent field and mark it as `sensitive`. But if you guys think the current implementation is OK. Then I am happy to approve.

Reply:
Waiting your last confirmation, then I will merge this PR.

/wait-any

Reply:
@wbpcode  We have discussed internally and we prefer to keep it as is. We already have the same situation in the HTTP exporter, where the token is also passed/used in the http headers, so we prefer to keep the config consistent to not confuse users.